tests/rptest/tests/consumer_group_balancing_test.py

```diff
@@ -7,195 +7,63 @@
-                                            group=group,
-                                            instance_id=instance_id)
-            consumer.start()
-            consumers_by_group[group] = [consumer]
-            consumers_all.append(consumer)
+        rpk.consume(topic=topic.name, n=1, group="test-group")
 
-        # wait for some messages
-        self.logger.info(f"waiting for some messages")
-        wait_until(
-            lambda: ConsumerGroupBalancingTest.consumed_at_least(
-                consumers_all, 50), 30, 2)
-        # all groups should be stable
-        self.logger.info(f"checking groups are stable")
-        for group in groups:
-            self.validate_group_state(group,
-                                      expected_state="Stable",
-                                      static_members=static_members,
-                                      group_members=1)
-        # stop consumers
-        self.logger.info(f"stopping producers and consumers")
-        for c in consumers_all:
-            c.stop()
-        for c in consumers_all:
-            c.wait()
-            c.free()
-        self.producer.wait()
-        self.producer.free()
+        admin = Admin(self.redpanda)
+        partitions = admin.get_partitions("__consumer_offsets")
+        replicas_per_node = defaultdict(lambda: 0)
 
-        rpk = RpkTool(self.redpanda)
-        coord_groups = {}
-        # find coordinator of each group
-        for group in groups:
-            group_desc = rpk.group_describe(group)
-            self.logger.info(f"group {group}: {group_desc}")
-            coord_groups[group_desc.coordinator] = coord_groups.get(
-                group_desc.coordinator, 0) + 1
+        expected_number_of_replicas = floor(
+            (consumer_group_topic_partitions * 3) / 5)
 
-        # model the ideal cgroup coordinators distribution
-        coord_density = sorted(coord_groups.values(), reverse=True)
-        expected_coord_density = [
-            ceil(len(groups) / len(self.redpanda.nodes))
-            for _ in range(len(groups) % len(self.redpanda.nodes))
-        ] + [
-            floor(len(groups) / len(self.redpanda.nodes)) for _ in range(
-                len(groups) %
-                len(self.redpanda.nodes), len(self.redpanda.nodes))
-        ]
+        for p in partitions:
+            for node in p['replicas']:
+                replicas_per_node[node['node_id']] += 1
 
-        # actual distribution must match the model
-        if coord_density != expected_coord_density:
-            self.logger.error(
-                "Uneven distribution of group coordinators across nodes. "
-                f"coordinators of groups: {coord_groups}, "
-                f"groups: {len(groups)}, brokers: {len(self.redpanda.nodes)}, "
-                f"coord_density: {coord_density}, expected_coord_density: {expected_coord_density}"
+        for node, replicas in replicas_per_node.items():
```

Comment:
We need a waiter for this? I thought the first assignment is 'random' but the balancer `eventually` balances by domain which is when actual uniformity is achieved.

Reply:
this is a data balance so it does it during the first assignment,